README.md

```diff
@@ -3986,5 +4206,8 @@ twine upload --repository pypi dist/* -u __token__ -p <PyPI_TOKEN>
 
 ### Development guides
 
-- [Slot rendering flot](https://github.com/EmilStenstrom/django-components/blob/master/docs/slot_rendering.md)
-- [Slots and blocks](https://github.com/EmilStenstrom/django-components/blob/master/docs/slots_and_blocks.md)
+Deep dive into how django_components' features are implemented.
+
+- [Slot rendering](https://github.com/EmilStenstrom/django-components/blob/master/docs/devguides/slot_rendering.md)
+- [Slots and blocks](https://github.com/EmilStenstrom/django-components/blob/master/docs/devguides/slots_and_blocks.md)
+- [JS and CSS dependency management](https://github.com/EmilStenstrom/django-components/blob/master/docs/devguides/dependency_mgmt.md)
```

Comment:
I added a doc on [how the JS / CSS dependency management works](https://github.com/EmilStenstrom/django-components/pull/688/files#diff-59a841a79c146e0f8f478fb78480b7930d6bf09da7fc59605ac4836c9527a6ef).

Also I moved these "dev guides" from `docs` to `docs/devguides`, so they are distinguishable from user-facing docs.